packages/dev/core/src/Meshes/meshUtilities.ts

```diff
@@ -0,0 +1,62 @@
+import { AnimationGroup } from "../Animations/animationGroup";
+import { BoundingBox } from "../Culling/boundingBox";
+import { Vector3 } from "../Maths/math.vector";
+import type { AbstractMesh } from "./abstractMesh";
+
+/**
+ * Computes the maximum bounding boxes of the given meshes taking animation, skeleton, morph targets into account.
+ * @param meshes The meshes to compute
+ * @param animationGroup An optional animation group to animate
+ * @param animationStep An optional animation step value indicating the amount to iterate while computing bounding boxes
+ * @returns An array of bounding boxes corresponding to the input meshes, animation group, and animation step values
+ */
+export function computeMaxBoundingBoxes(meshes: Array<AbstractMesh>, animationGroup?: AnimationGroup, animationStep = 1 / 6): Array<BoundingBox> {

```

Comment:
If no `animationGroup` is provided, then no `animationStep` should be provided, correct? Would it be worth it to reflect this in the API, either with classic overloads or named tuples?
```suggestion
export function computeMaxBoundingBoxes(...args: [meshes: Array<AbstractMesh>] | [meshes: Array<AbstractMesh>, animationGroup: AnimationGroup] | [meshes: Array<AbstractMesh>, animationGroup: AnimationGroup, animationStep: number]): Array<BoundingBox> {

```

Reply:
I personally don't think this is worth the effort. It makes the code more difficult to read.

I considered using a `options` arguments instead of individual arguments but decided against it also.

No strong preference though.

Reply:
Personally I would prefer an `options` object using discriminated unions to make it super clear that it is not valid to provide an animationStep without providing an animationGroup. I think this kind of API design just helps ppl understand how to use it, but your call.

Reply:
I think requiring that `animationStep` not be passed in when `animationGroup` is not present is harder to use. Most of the time, the user doesn't know if there is going to be an animationGroup or not.